XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -96,6 +97,39 @@ public NavigationRootPage()
             CoreApplication.GetCurrentView().TitleBar.LayoutMetricsChanged += (s, e) => UpdateAppTitle(s);
 
             _isKeyboardConnected = Convert.ToBoolean(new KeyboardCapabilities().KeyboardPresent);
+
+
+            // remove the solid-colored backgrounds behind the caption controls and system back button
+            // This is done when the app is loaded since before that the actual theme that is used is not "determined" yet
+            Loaded += delegate (object sender, RoutedEventArgs e) {
+                ApplicationViewTitleBar titleBar = ApplicationView.GetForCurrentView().TitleBar;
+                titleBar.ButtonBackgroundColor = Colors.Transparent;
+                titleBar.ButtonInactiveBackgroundColor = Colors.Transparent;
+
+                var currentTheme = App.RootTheme.ToString();
+                bool darkTheme = false;
+
+                switch (currentTheme)
+                {
+                    case "Dark":
+                        darkTheme = true;
+                        break;
+                    case "Default":
+                        if (Application.Current.RequestedTheme == ApplicationTheme.Dark)
+                        {
+                            darkTheme = true;
+                        }
+                        break;
+                }
+                if (darkTheme)
+                {
+                    titleBar.ButtonForegroundColor = Colors.White;
```

Comment:
I surprised that these explicit color declarations (still) work in high contrast and behave as expected. Perhaps the API is clever enough to disallow changing the ButtonForegroundColor property when high contrast is enabled. I'm not exactly sure why, but everything is working well :)